🐛 Project with a custom dev client is failing during build

### What were you trying to do?

I am trying to install the library in my custom dev client project. 
I followed the following instructions: 
1. expo install react-native-vision-camera
2. added the following config plugin to my app.json
"plugins": [
    [
      "react-native-vision-camera",
      {
        "cameraPermissionText": "EventShare needs access to your Camera.",

        // optionally, if you want to record audio:
        "enableMicrophonePermission": true,
        "microphonePermissionText": "EventShare needs access to your Microphone."
      }
    ]
  ]
3. rebuilt the binary using
      **eas build --profile development --platform all**


### What happened instead?

The build fails with the following error on Android  build from EAS build: 
```
[stderr] FAILURE: Build failed with an exception.
[stderr] * What went wrong:
[stderr] Execution failed for task ':react-native-vision-camera:externalNativeBuildDebug'.
[stderr] > Build command failed.
[stderr]   Error while executing process /opt/android/cmake/3.10.2.4988404/bin/ninja with arguments {-C /build/workingdir/build/node_modules/react-native-vision-camera/android/.cxx/cmake/debug/armeabi-v7a VisionCamera}
[stderr]   ninja: Entering directory `/build/workingdir/build/node_modules/react-native-vision-camera/android/.cxx/cmake/debug/armeabi-v7a'
[stderr]   
[stderr]   ninja: error: '../../../../build/third-party-ndk/hermes/jni/armeabi-v7a/libhermes.so', needed by '../../../../build/intermediates/cmake/debug/obj/armeabi-v7a/libVisionCamera.so', missing and no known rule to make it
[stderr] * Try:
[stderr] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 6m 35s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings
875 actionable tasks: 875 executed
Error: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.
```

following error on iOS build from EAS build: 


```
❌  (ios/Pods/Headers/Public/ReactCommon/ReactCommon/TurboModuleUtils.h:13:10)

  11 | #include <string>
  12 | 
> 13 | #include <folly/Optional.h>
     |          ^ 'folly/Optional.h' file not found
  14 | #include <jsi/jsi.h>
  15 | 
  16 | #include <ReactCommon/CallInvoker.h>
› Compiling react-native-vision-camera Pods/VisionCamera » JSIUtils.mm
› Compiling react-native-vision-camera Pods/VisionCamera » JSConsoleHelper.mm
› Compiling react-native-vision-camera Pods/VisionCamera » FrameProcessorUtils.mm
› Compiling react-native-vision-camera Pods/VisionCamera » FrameProcessorRuntimeManager.mm
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/EventShare-cvvxuvgbosgptrfgtqzckczinmol/Build/Intermediates.noindex/ArchiveIntermediates/EventShare/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/VisionCamera.build/Objects-normal/arm64/RCTBridge+runOnJS.o /Users/expo/workingdir/build/node_modules/react-native-vision-camera/ios/React\ Utils/RCTBridge+runOnJS.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
▸ (1 failure)
** ARCHIVE FAILED **
The following build commands failed:
	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/EventShare-cvvxuvgbosgptrfgtqzckczinmol/Build/Intermediates.noindex/ArchiveIntermediates/EventShare/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/VisionCamera.build/Objects-normal/arm64/RCTBridge+runOnJS.o /Users/expo/workingdir/build/node_modules/react-native-vision-camera/ios/React\ Utils/RCTBridge+runOnJS.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
Exit status: 65
+-------------+-------------------------+
|           Build environment           |
+-------------+-------------------------+
| xcode_path  | /Applications/Xcode.app |
| gym_version | 2.185.1                 |
| sdk         | iPhoneOS14.5.sdk        |
+-------------+-------------------------+
Looks like fastlane ran into a build/archive error with your project
It's hard to tell what's causing the error, so we wrote some guides on how
to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
Before submitting an issue on GitHub, please follow the guide above and make
sure your project is set up correctly.
fastlane uses `xcodebuild` commands to generate your binary, you can see the
the full commands printed out in yellow in the above log.

```

### Relevant log output

```shell
_______________________________________ANDROID LOGS__________________________________________
> Task :react-native-vision-camera:extractAARHeaders
> Task :react-native-vision-camera:extractJNIFiles
> Task :react-native-vision-camera:generateJsonModelDebug
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|armeabi-v7a : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|armeabi-v7a : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|arm64-v8a : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|arm64-v8a : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|x86 : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|x86 : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|x86_64 : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
/build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|x86_64 : CMake Warning at /build/workingdir/build/node_modules/react-native-vision-camera/android/CMakeLists.txt:122 (message):
  VisionCamera linking: FOR_HERMES=true
> Task :react-native-vision-camera:prepareHermes UP-TO-DATE
> Task :app:compileDebugJavaWithJavac
> Task :app:compileDebugSources
[stderr] Note: /build/workingdir/build/android/app/src/debug/java/com/simplyrem/GalleryShare/ReactNativeFlipper.java uses or overrides a deprecated API.
[stderr] Note: Recompile with -Xlint:deprecation for details.
> Task :react-native-vision-camera:prepareJSC
> Task :app:mergeDebugJavaResource
> Task :react-native-vision-camera:createNativeDepsDirectories UP-TO-DATE
> Task :react-native-vision-camera:downloadBoost
Download https://github.com/react-native-community/boost-for-react-native/releases/download/v1.63.0-0/boost_1_63_0.tar.gz
> Task :app:dexBuilderDebug
> Task :app:mergeLibDexDebug
> Task :app:mergeProjectDexDebug
> Task :app:mergeDebugJniLibFolders
> Task :unimodules-app-loader:mergeDebugJniLibFolders
> Task :unimodules-app-loader:mergeDebugNativeLibs NO-SOURCE
> Task :unimodules-app-loader:stripDebugDebugSymbols NO-SOURCE
> Task :unimodules-app-loader:copyDebugJniLibsProjectOnly
> Task :unimodules-core:mergeDebugJniLibFolders
> Task :unimodules-core:mergeDebugNativeLibs NO-SOURCE
> Task :unimodules-core:stripDebugDebugSymbols NO-SOURCE
> Task :unimodules-core:copyDebugJniLibsProjectOnly
> Task :unimodules-react-native-adapter:mergeDebugJniLibFolders
> Task :unimodules-react-native-adapter:mergeDebugNativeLibs NO-SOURCE
> Task :unimodules-react-native-adapter:stripDebugDebugSymbols NO-SOURCE
> Task :unimodules-react-native-adapter:copyDebugJniLibsProjectOnly
> Task :unimodules-task-manager-interface:mergeDebugJniLibFolders
> Task :unimodules-task-manager-interface:mergeDebugNativeLibs NO-SOURCE
> Task :unimodules-task-manager-interface:stripDebugDebugSymbols NO-SOURCE
> Task :unimodules-task-manager-interface:copyDebugJniLibsProjectOnly
> Task :app:validateSigningDebug
> Task :react-native-vision-camera:prepareBoost
> Task :react-native-vision-camera:downloadDoubleConversion
Download https://github.com/google/double-conversion/archive/v1.1.6.tar.gz
> Task :react-native-vision-camera:prepareDoubleConversion
> Task :react-native-vision-camera:downloadFolly
Download https://github.com/facebook/folly/archive/v2020.01.13.00.tar.gz
> Task :react-native-vision-camera:prepareFolly
> Task :react-native-vision-camera:downloadGlog
Download https://github.com/google/glog/archive/v0.3.5.tar.gz
> Task :react-native-vision-camera:prepareGlog
> Task :react-native-vision-camera:prepareThirdPartyNdkHeaders
> Task :react-native-vision-camera:externalNativeBuildDebug FAILED
Build VisionCamera_armeabi-v7a
ninja: Entering directory `/build/workingdir/build/node_modules/react-native-vision-camera/android/.cxx/cmake/debug/armeabi-v7a'
ninja: error: '../../../../build/third-party-ndk/hermes/jni/armeabi-v7a/libhermes.so', needed by '../../../../build/intermediates/cmake/debug/obj/armeabi-v7a/libVisionCamera.so', missing and no known rule to make it
w: Detected multiple Kotlin daemon sessions at build/kotlin/sessions
[stderr] FAILURE: Build failed with an exception.
[stderr] * What went wrong:
[stderr] Execution failed for task ':react-native-vision-camera:externalNativeBuildDebug'.
[stderr] > Build command failed.
[stderr]   Error while executing process /opt/android/cmake/3.10.2.4988404/bin/ninja with arguments {-C /build/workingdir/build/node_modules/react-native-vision-camera/android/.cxx/cmake/debug/armeabi-v7a VisionCamera}
[stderr]   ninja: Entering directory `/build/workingdir/build/node_modules/react-native-vision-camera/android/.cxx/cmake/debug/armeabi-v7a'
[stderr]   
[stderr]   ninja: error: '../../../../build/third-party-ndk/hermes/jni/armeabi-v7a/libhermes.so', needed by '../../../../build/intermediates/cmake/debug/obj/armeabi-v7a/libVisionCamera.so', missing and no known rule to make it
[stderr] * Try:
[stderr] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 6m 35s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings
875 actionable tasks: 875 executed



__________________________________________IOS LOGS___________________________________________
› Compiling react-native-vision-camera Pods/VisionCamera » AVAssetWriter.Status+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVAssetWriterInputPixelBufferAdaptor+initWithVideoSettings.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVAudioSession+trySetAllowHaptics.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVAudioSession+updateCategory.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVAuthorizationStatus+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureColorSpace+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureConnection+setInterfaceOrientation.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice+isMultiCam.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice+neutralZoom.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice+physicalDevices.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.DeviceType+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.FlashMode+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Format+isBetterThan.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Format+matchesFilter.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Format+toDictionary.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Format+videoDimensions.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Format.AutoFocusSystem+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.Position+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureDevice.TorchMode+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCapturePhotoOutput+mirror.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCapturePhotoOutput.QualityPrioritization+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureSession+setVideoStabilizationMode.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureSession.Preset+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVCaptureVideoStabilizationMode+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVFileType+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVFrameRateRange+includes.swift
› Compiling react-native-vision-camera Pods/VisionCamera » AVVideoCodecType+descriptor.swift
› Compiling react-native-vision-camera Pods/VisionCamera » Callback.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraError.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraQueues.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+AVAudioSession.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+AVCaptureSession.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+Focus.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+RecordVideo.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+TakePhoto.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView+Zoom.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraView.swift
› Compiling react-native-vision-camera Pods/VisionCamera » CameraViewManager.swift
› Compiling react-native-vision-camera Pods/VisionCamera » Collection+safe.swift
› Compiling react-native-vision-camera Pods/VisionCamera » EnumParserError.swift
› Compiling react-native-vision-camera Pods/VisionCamera » MakeReactError.swift
› Compiling react-native-vision-camera Pods/VisionCamera » PhotoCaptureDelegate.swift
› Compiling react-native-vision-camera Pods/VisionCamera » Promise.swift
› Compiling react-native-vision-camera Pods/VisionCamera » ReactLogger.swift
› Compiling react-native-vision-camera Pods/VisionCamera » RecordingSession.swift
› Packaging react-native Pods/React-CoreModules » libReact-CoreModules.a
› Compiling react-native-static-safe-area-insets Pods/RNStaticSafeAreaInsets » RNStaticSafeAreaInsets.m
› Packaging @react-native-community/blur Pods/react-native-blur » libreact-native-blur.a
› Compiling react-native-static-safe-area-insets Pods/RNStaticSafeAreaInsets » RNStaticSafeAreaInsets-dummy.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementTransitionManager.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementTransitionItem.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementTransition.m
› Packaging react-native-static-safe-area-insets Pods/RNStaticSafeAreaInsets » libRNStaticSafeAreaInsets.a
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementStyle.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementNodeManager.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementNode.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementCornerRadii.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElementContent.m
› Compiling react-native-shared-element Pods/RNSharedElement » RNSharedElement-dummy.m
› Packaging react-native-screens Pods/RNScreens » libRNScreens.a
› Compiling react-native-svg Pods/RNSVG » RNSVGViewBox.m
› Compiling react-native-svg Pods/RNSVG » RNSVGUseManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGUse.m
› Packaging react-native-shared-element Pods/RNSharedElement » libRNSharedElement.a
› Compiling react-native-svg Pods/RNSVG » RNSVGTopAlignedLabel.ios.m
› Compiling react-native-svg Pods/RNSVG » RNSVGTextProperties.m
› Compiling react-native-svg Pods/RNSVG » RNSVGTextPathManager.m
› Compiling react-native-svg RNSVGTextPath.m
› Compiling react-native-svg Pods/RNSVG » RNSVGTextManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGText.m
› Compiling react-native-svg Pods/RNSVG » RNSVGTSpanManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGTSpan.m
› Compiling react-native-svg Pods/RNSVG » RNSVGSymbolManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGSymbol.m
› Compiling react-native-svg Pods/RNSVG » RNSVGSvgViewManager.m
› Packaging react-native-view-shot Pods/react-native-view-shot » libreact-native-view-shot.a
› Compiling react-native-svg Pods/RNSVG » RNSVGSvgView.m
› Compiling react-native-svg Pods/RNSVG » RNSVGSolidColorBrush.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRenderableManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRenderable.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRectManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRect.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRadialGradientManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGRadialGradient.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPropHelper.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPatternManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPattern.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPathParser.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPathMeasure.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPathManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPath.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPainterBrush.m
› Compiling react-native-svg Pods/RNSVG » RNSVGPainter.m
› Compiling react-native-svg Pods/RNSVG » RNSVGNodeManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGNode.m
› Compiling react-native-svg Pods/RNSVG » RNSVGMaskManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGMask.m
› Compiling react-native-svg Pods/RNSVG » RNSVGMarkerPosition.m
› Compiling react-native-svg Pods/RNSVG » RNSVGMarkerManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGMarker.m
› Compiling react-native-svg Pods/RNSVG » RNSVGLinearGradientManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGLinearGradient.m
› Compiling react-native-svg Pods/RNSVG » RNSVGLineManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGLine.m
› Compiling react-native-svg Pods/RNSVG » RNSVGLength.m
› Compiling react-native-svg Pods/RNSVG » RNSVGImageManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGImage.m
› Compiling react-native-svg Pods/RNSVG » RNSVGGroupManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGGroup.m
› Compiling react-native-svg Pods/RNSVG » RNSVGGlyphContext.m
› Compiling react-native-svg Pods/RNSVG » RNSVGForeignObjectManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGForeignObject.m
› Compiling react-native-svg Pods/RNSVG » RNSVGFontData.m
› Compiling react-native-svg Pods/RNSVG » RNSVGEllipseManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGDefsManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGDefs.m
› Compiling react-native-svg Pods/RNSVG » RNSVGContextBrush.m
› Compiling react-native-svg Pods/RNSVG » RNSVGClipPathManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGClipPath.m
› Compiling react-native-svg Pods/RNSVG » RNSVGCircleManager.m
› Compiling react-native-svg Pods/RNSVG » RNSVGCircle.m
› Compiling react-native-svg Pods/RNSVG » RNSVGBrush.m
› Compiling react-native-svg Pods/RNSVG » RNSVGBezierElement.m
› Compiling react-native-svg Pods/RNSVG » RNSVG-dummy.m
› Compiling react-native-svg Pods/RNSVG » RCTConvert+RNSVG.m
› Compiling @react-native-community/masked-view Pods/RNCMaskedView » RNCMaskedViewManager.m
› Compiling @react-native-community/masked-view Pods/RNCMaskedView » RNCMaskedView.m
› Compiling @react-native-community/masked-view Pods/RNCMaskedView » RNCMaskedView-dummy.m
› Packaging react-native-svg Pods/RNSVG » libRNSVG.a
› Compiling react-native-reanimated Pods/RNReanimated » WorkletsCache.cpp
› Compiling react-native-reanimated Pods/RNReanimated » WorkletEventHandler.cpp
› Compiling react-native-reanimated Pods/RNReanimated » UIResponder+Reanimated.mm
› Compiling react-native-reanimated Pods/RNReanimated » ShareableValue.cpp
› Compiling react-native-reanimated Pods/RNReanimated » Scheduler.cpp
› Compiling react-native-reanimated Pods/RNReanimated » RuntimeDecorator.cpp
› Compiling react-native-reanimated Pods/RNReanimated » RemoteObject.cpp
› Compiling react-native-reanimated Pods/RNReanimated » RNReanimated-dummy.m
› Compiling react-native-reanimated Pods/RNReanimated » REAValueNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REATransitionValues.m
› Compiling react-native-reanimated Pods/RNReanimated » REATransitionManager.m
› Compiling react-native-reanimated Pods/RNReanimated » REATransitionAnimation.m
› Compiling react-native-reanimated Pods/RNReanimated » REATransition.m
› Compiling react-native-reanimated Pods/RNReanimated » REATransformNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAStyleNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAPropsNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAParamNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAOperatorNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REANodesManager.m
› Compiling react-native-reanimated Pods/RNReanimated » REANode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAModule.m
› Compiling react-native-reanimated Pods/RNReanimated » REAJSCallNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAIOSScheduler.mm
› Compiling react-native-reanimated Pods/RNReanimated » REAIOSLogger.mm
› Compiling react-native-reanimated Pods/RNReanimated » REAIOSErrorHandler.mm
› Compiling react-native-reanimated Pods/RNReanimated » REAFunctionNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAEventNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAEventDispatcher.m
› Compiling react-native-reanimated Pods/RNReanimated » READebugNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REACondNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAConcatNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAClockNodes.m
› Compiling react-native-reanimated Pods/RNReanimated » REACallFuncNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REABlockNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REABezierNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAAlwaysNode.m
› Compiling react-native-reanimated Pods/RNReanimated » REAAllTransitions.m
› Compiling react-native-reanimated Pods/RNReanimated » RCTConvert+REATransition.m
› Compiling react-native-reanimated Pods/RNReanimated » NativeReanimatedModuleSpec.cpp
› Compiling react-native-reanimated Pods/RNReanimated » NativeReanimatedModule.cpp
› Compiling react-native-reanimated Pods/RNReanimated » NativeProxy.mm
› Compiling react-native-reanimated Pods/RNReanimated » NativeMethods.mm
› Compiling react-native-reanimated Pods/RNReanimated » MutableValueSetterProxy.cpp
› Compiling react-native-reanimated Pods/RNReanimated » MutableValue.cpp
› Compiling react-native-reanimated Pods/RNReanimated » MapperRegistry.cpp
› Compiling react-native-reanimated Pods/RNReanimated » Mapper.cpp
› Compiling react-native-vision-camera Pods/VisionCamera » VisionCameraScheduler.mm
› Packaging expo-dev-menu-interface Pods/expo-dev-menu-interface » libexpo-dev-menu-interface.a
› Compiling react-native-vision-camera Pods/VisionCamera » VisionCamera-dummy.m
› Compiling react-native-vision-camera Pods/VisionCamera » RCTBridge+runOnJS.mm

❌  (ios/Pods/Headers/Public/ReactCommon/ReactCommon/TurboModuleUtils.h:13:10)

  11 | #include <string>
  12 | 
> 13 | #include <folly/Optional.h>
     |          ^ 'folly/Optional.h' file not found
  14 | #include <jsi/jsi.h>
  15 | 
  16 | #include <ReactCommon/CallInvoker.h>
› Compiling react-native-vision-camera Pods/VisionCamera » JSIUtils.mm
› Compiling react-native-vision-camera Pods/VisionCamera » JSConsoleHelper.mm
› Compiling react-native-vision-camera Pods/VisionCamera » FrameProcessorUtils.mm
› Compiling react-native-vision-camera Pods/VisionCamera » FrameProcessorRuntimeManager.mm
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/EventShare-cvvxuvgbosgptrfgtqzckczinmol/Build/Intermediates.noindex/ArchiveIntermediates/EventShare/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/VisionCamera.build/Objects-normal/arm64/RCTBridge+runOnJS.o /Users/expo/workingdir/build/node_modules/react-native-vision-camera/ios/React\ Utils/RCTBridge+runOnJS.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
▸ (1 failure)
** ARCHIVE FAILED **
The following build commands failed:
	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/EventShare-cvvxuvgbosgptrfgtqzckczinmol/Build/Intermediates.noindex/ArchiveIntermediates/EventShare/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/VisionCamera.build/Objects-normal/arm64/RCTBridge+runOnJS.o /Users/expo/workingdir/build/node_modules/react-native-vision-camera/ios/React\ Utils/RCTBridge+runOnJS.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
Exit status: 65
+-------------+-------------------------+
|           Build environment           |
+-------------+-------------------------+
| xcode_path  | /Applications/Xcode.app |
| gym_version | 2.185.1                 |
| sdk         | iPhoneOS14.5.sdk        |
+-------------+-------------------------+
Looks like fastlane ran into a build/archive error with your project
It's hard to tell what's causing the error, so we wrote some guides on how
to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
Before submitting an issue on GitHub, please follow the guide above and make
sure your project is set up correctly.
fastlane uses `xcodebuild` commands to generate your binary, you can see the
the full commands printed out in yellow in the above log.
Make sure to inspect the output above, as usually you'll find more error information there
[stderr] [!] Error building the application - see the log above
Error: Fastlane build failed with unknown error. Please refer to the "Run fastlane" and "Xcode Logs" phases.
Fastlane errors in most cases are not printed at the end of the output, so you may not find any useful information in the last lines of output when looking for an error message.
```


### Device

iPhone 12

### VisionCamera Version

2.5.0

### Additional information

- [X] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -10,6 +10,17 @@ set (RN_SO_DIR ${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/first-
 
 # VisionCamera shared
 
+if(${REACT_NATIVE_VERSION} LESS 66)
+        set (
+                INCLUDE_JSI_CPP
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi/jsi/jsi.cpp"
+        )
+        set (
+                INCLUDE_JSIDYNAMIC_CPP
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi/jsi/JSIDynamic.cpp"
+        )
+endif()
+
 add_library(
         ${PACKAGE_NAME}
         SHARED
@@ -31,11 +42,14 @@ file (GLOB LIBFBJNI_INCLUDE_DIR "${BUILD_DIR}/fbjni-*-headers.jar/")
 target_include_directories(
         ${PACKAGE_NAME}
         PRIVATE
+        # --- fbjni ---
         "${LIBFBJNI_INCLUDE_DIR}"
+        # --- Third Party (required by RN) ---
         "${BUILD_DIR}/third-party-ndk/boost"
         "${BUILD_DIR}/third-party-ndk/double-conversion"
         "${BUILD_DIR}/third-party-ndk/folly"
         "${BUILD_DIR}/third-party-ndk/glog"
+        # --- React Native ---
         "${NODE_MODULES_DIR}/react-native/React"
         "${NODE_MODULES_DIR}/react-native/React/Base"
         "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni"
@@ -44,6 +58,9 @@ target_include_directories(
         "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
         "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
         "${NODE_MODULES_DIR}/hermes-engine/android/include/"
+        ${INCLUDE_JSI_CPP} # only on older RN versions
+        ${INCLUDE_JSIDYNAMIC_CPP} # only on older RN versions
+        # --- Reanimated ---
         "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/Tools"
         "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SpecTools"
         "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SharedItems"
@@ -88,12 +105,6 @@ find_library(
         NO_CMAKE_FIND_ROOT_PATH
 )
 find_library(
-        JSI_LIB
-        jsi
-        PATHS ${LIBRN_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
-find_library(
         FOLLY_JSON_LIB
         folly_json
         PATHS ${LIBRN_DIR}
@@ -106,6 +117,18 @@ find_library(
         PATHS ${LIBRN_DIR}
         NO_CMAKE_FIND_ROOT_PATH
 )
+if(${REACT_NATIVE_TARGET_VERSION} LESS 66)
+        # JSI lib didn't exist on RN 0.65 and before. Simply omit it.
+        set (JSI_LIB "")
+else()
+        # RN 0.66 distributes libjsi.so, can be used instead of compiling jsi.cpp manually.
+        find_library(
+                JSI_LIB
+                jsi
+                PATHS ${LIBRN_DIR}
+                NO_CMAKE_FIND_ROOT_PATH
+        )
+endif()
 
 find_library(
         REANIMATED_LIB

--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -117,7 +117,7 @@ find_library(
         PATHS ${LIBRN_DIR}
         NO_CMAKE_FIND_ROOT_PATH
 )
-if(${REACT_NATIVE_TARGET_VERSION} LESS 66)
+if(${REACT_NATIVE_VERSION} LESS 66)
         # JSI lib didn't exist on RN 0.65 and before. Simply omit it.
         set (JSI_LIB "")
 else()

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -118,6 +118,8 @@ android {
   packagingOptions {
     // Exclude all Libraries that are already present in the user's app (through React Native or by him installing REA)
     excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libjsi.so", "**/libreactnativejni.so", "**/libfolly_json.so", "**/libreanimated.so", "**/libjscexecutor.so"]
+    // META-INF is duplicate by CameraX.
+    exclude "META-INF/**"
   }
 
   buildTypes {

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -116,7 +116,8 @@ android {
   }
 
   packagingOptions {
-    excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libjsi.so"]
+    // Exclude all Libraries that are already present in the user's app (through React Native or by him installing REA)
+    excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libjsi.so", "**/libreactnativejni.so", "**/libfolly_json.so", "**/libreanimated.so", "**/libjscexecutor.so"]
   }
 
   buildTypes {

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -4,6 +4,10 @@ import java.nio.file.Paths
 
 def reactNative = new File("$projectDir/../node_modules/react-native")
 
+def reactProperties = new Properties()
+file("$projectDir/../node_modules/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
+def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME").split("\\.")[1].toInteger()
+
 def FOR_HERMES = System.getenv("FOR_HERMES") == "True"
 rootProject.getSubprojects().forEach({project ->
   if (project.plugins.hasPlugin("com.android.application")) {
@@ -94,6 +98,7 @@ android {
         cppFlags "-fexceptions", "-frtti", "-std=c++1y", "-DONANDROID"
         abiFilters 'x86', 'x86_64', 'armeabi-v7a', 'arm64-v8a'
         arguments '-DANDROID_STL=c++_shared',
+                "-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION}",
                 "-DNODE_MODULES_DIR=${rootDir}/../node_modules",
                 "-DFOR_HERMES=${FOR_HERMES}"
       }

